src/js/actions/transform.js

```diff
@@ -1317,16 +1317,36 @@ define(function (require, exports) {
 
             return Promise.join(modelPromise, transformPromise);
         } else {
+            var dragUpdatePromise,
+                dragLayerIDs = this.flux.store("tool").getState().draggedLayerIDs;
+
+            if (dragLayerIDs) {
+                var dragLayers = dragLayerIDs.map(currentDoc.layers.byID.bind(currentDoc.layers));
+                dragUpdatePromise = this.transfer("layers.select", currentDoc, dragLayers);
+            } else {
+                dragUpdatePromise = Promise.resolve();
+            }
+
             // short circuit based on this trackerEndedWithoutBreakingHysteresis event flag
             if (event.trackerEndedWithoutBreakingHysteresis) {
-                return Promise.resolve();
+                return dragUpdatePromise;
             } else {
-                // The history event may arrive before or after the transform event. The
-                // following ensures correct behavior w.r.t. history amendment by reset
-                // actions for either case.
-                return this.transfer("history.newHistoryStateRogueSafe", currentDoc.id)
+                return dragUpdatePromise
                     .bind(this)
                     .then(function () {
+                        // This doesn't cause a re-render
+                        this.dispatch(events.tool.SELECT_TOOL_DRAG, { selectedIDs: null });
+
+                        // The history event may arrive before or after the transform event. The
+                        // following ensures correct behavior w.r.t. history amendment by reset
+                        // actions for either case.
+                        return this.transfer("history.newHistoryStateRogueSafe", currentDoc.id);
```

Comment:
This seems fine.  By delaying this in time until after the layer selection is complete, that might increase the chance that the "rogue safe" logic is triggered ... which is to say that this increases the likelihood that the historyState event will be handled before this history action is ... and that's OK.  I can't think of a strong argument for moving this action earlier.  

The layer selection should not affect the history state.